### ISSUE TITLE
Unity Crash when run demo

### ISSUE THREAD

**Issue by codingwatching** [crashkonijn/GOAP#15]
Hi @crashkonijn 
I found this project and it is really awesome! I like this so much and I think can apply this to make some funny bots for my game.
Recently, when I run the demo on a Macbook Pro, chip M1 Max, Unity 2022.2.11f1 Silicon version
The Editor Crash and the log is like this
```
Obtained 91 stack frames.
#0  0x00000100bde960 in ScriptableRenderLoopShadowsJob(GfxDeviceAsyncCommand::ArgScratch*, GfxDeviceAsyncCommand::Arg const*)
#1  0x00000100f3d7a4 in GfxDevice::ExecuteAsync(int, void (*)(GfxDeviceAsyncCommand::ArgScratch*, GfxDeviceAsyncCommand::Arg const*), GfxDeviceAsyncCommand::ArgScratch**, GfxDeviceAsyncCommand::Arg const*, JobFence const&)
#2  0x00000100bd5c20 in ScheduleRenderJobs(unsigned int, ScriptableRenderLoopShadowsArg const*, JobFence const&)
#3  0x00000100bd5828 in ExecuteDrawShadowsCommand(SlicedDrawShadow const&)
#4  0x00000100bdb734 in ScriptableRenderContext::ExecuteScriptableRenderLoop()
#5  0x0000035d1d4774 in  (wrapper managed-to-native) UnityEngine.Rendering.ScriptableRenderContext:Submit_Internal_Injected (UnityEngine.Rendering.ScriptableRenderContext&) [{0x3276be998} + 0xa4]  (0x35d1d46d0 0x35d1d4808) [0x12ff96a80 - Unity Child Domain]
#6  0x0000035d1d4664 in  UnityEngine.Rendering.ScriptableRenderContext:Submit_Internal () [{0x3276be8b0} + 0x3c]  (0x35d1d4628 0x35d1d4680) [0x12ff96a80 - Unity Child Domain]
#7  0x0000035d1d45a0 in  UnityEngine.Rendering.ScriptableRenderContext:Submit () [{0x3276294b0} + 0x78] [/Users/bokken/build/output/unity/unity/Runtime/Export/RenderPipeline/ScriptableRenderContext.cs :: 117u] (0x35d1d4528 0x35d1d45c8) [0x12ff96a80 - Unity Child Domain]
#8  0x0000035d11353c in  UnityEngine.Rendering.Universal.UniversalRenderPipeline:RenderSingleCamera (UnityEngine.Rendering.ScriptableRenderContext,UnityEngine.Rendering.Universal.CameraData&,bool) [{0x3275f4ad0} + 0x1014] [./Library/PackageCache/com.unity.render-pipelines.universal@14.0.6/Runtime/UniversalRenderPipeline.cs :: 657u] (0x35d112528 0x35d113604) [0x12ff96a80 - Unity Child Domain]
#9  0x0000035d107eb4 in  UnityEngine.Rendering.Universal.UniversalRenderPipeline:RenderSingleCameraInternal (UnityEngine.Rendering.ScriptableRenderContext,UnityEngine.Camera) [{0x3275f4758} + 0x2a4] [./Library/PackageCache/com.unity.render-pipelines.universal@14.0.6/Runtime/UniversalRenderPipeline.cs :: 527u] (0x35d107c10 0x35d107f00) [0x12ff96a80 - Unity Child Domain]
#10 0x0000035d0f4704 in  UnityEngine.Rendering.Universal.UniversalRenderPipeline:Render (UnityEngine.Rendering.ScriptableRenderContext,System.Collections.Generic.List`1<UnityEngine.Camera>) [{0x348d4e710} + 0x8dc] [./Library/PackageCache/com.unity.render-pipelines.universal@14.0.6/Runtime/UniversalRenderPipeline.cs :: 369u] (0x35d0f3e28 0x35d0f4b14) [0x12ff96a80 - Unity Child Domain]
#11 0x0000035d0f3654 in  UnityEngine.Rendering.RenderPipeline:InternalRender (UnityEngine.Rendering.ScriptableRenderContext,System.Collections.Generic.List`1<UnityEngine.Camera>) [{0x2ec0c2a98} + 0x154] [/Users/bokken/build/output/unity/unity/Runtime/Export/RenderPipeline/RenderPipeline.cs :: 52u] (0x35d0f3500 0x35d0f367c) [0x12ff96a80 - Unity Child Domain]
#12 0x0000035d052308 in  UnityEngine.Rendering.RenderPipelineManager:DoRenderLoop_Internal (UnityEngine.Rendering.RenderPipelineAsset,intptr,UnityEngine.Object,Unity.Collections.LowLevel.Unsafe.AtomicSafetyHandle) [{0x2ecb4f8b8} + 0x220] [/Users/bokken/build/output/unity/unity/Runtime/Export/RenderPipeline/RenderPipelineManager.cs :: 126u] (0x35d0520e8 0x35d052448) [0x12ff96a80 - Unity Child Domain]
#13 0x0000035d0527b4 in  (wrapper runtime-invoke) <Module>:runtime_invoke_void_object_intptr_object_AtomicSafetyHandle (object,intptr,intptr,intptr) [{0x2ec0c2c80} + 0x12c]  (0x35d052688 0x35d0528dc) [0x12ff96a80 - Unity Child Domain]
#14 0x0000016b094884 in mono_jit_runtime_invoke
#15 0x0000016b21b480 in do_runtime_invoke
#16 0x0000016b21b3a0 in mono_runtime_invoke
#17 0x0000010103fc34 in scripting_method_invoke(ScriptingMethodPtr, ScriptingObjectPtr, ScriptingArguments&, ScriptingExceptionPtr*, bool)
#18 0x0000010101ee40 in ScriptingInvocation::Invoke(ScriptingExceptionPtr*, bool)
#19 0x00000100bdcb74 in ScriptableRenderContext::ExtractAndExecuteRenderPipeline(dynamic_array<Camera*, 0ul> const&, void (*)(SceneNode const*, AABB const*, IndexList&, SceneCullingParameters const*), void*, ScriptingObjectPtr)
#20 0x00000102cf2ed8 in Camera::ExecuteCustomRenderPipeline(Camera::EditorDrawingMode, DrawGridParameters const*, Camera::RenderFlag, core::hash_set<int, core::hash<int>, std::__1::equal_to<int> > const&, core::hash_set<int, core::hash<int>, std::__1::equal_to<int> > const&, ColorRGBAf, ColorRGBAf, float, float, int, Camera::OutlineDrawMode)
#21 0x00000102cf3b18 in Camera::RenderEditorCamera(Camera::EditorDrawingMode, DrawGridParameters const*, CullResults*, Camera::RenderFlag, int, bool, bool, core::hash_set<int, core::hash<int>, std::__1::equal_to<int> > const&, core::hash_set<int, core::hash<int>, std::__1::equal_to<int> > const&, core::hash_set<GameObject*, core::hash<GameObject*>, std::__1::equal_to<GameObject*> >*, ColorRGBAf, ColorRGBAf, float, float, int, Camera::OutlineDrawMode)
#22 0x00000102cf5758 in Camera::RenderEditorCamera(Camera::EditorDrawingMode, DrawGridParameters const*, bool, bool, core::hash_set<GameObject*, core::hash<GameObject*>, std::__1::equal_to<GameObject*> >*, CullResults*, std::__1::set<GameObject*, std::__1::less<GameObject*>, stl_allocator<GameObject*, (MemLabelIdentifier)1, 16> >*, ColorRGBAf, ColorRGBAf, float, float, int, Camera::OutlineDrawMode)
#23 0x000001006e1fb4 in Handles_CUSTOM_Internal_DrawCameraWithGrid(ScriptingBackendNativeObjectPtrOpaque*, Camera::EditorDrawingMode, DrawGridParameters&, unsigned char, unsigned char)
#24 0x0000035d05200c in  (wrapper managed-to-native) UnityEditor.Handles:Internal_DrawCameraWithGrid (UnityEngine.Camera,UnityEditor.DrawCameraMode,UnityEditor.DrawGridParameters&,bool,bool) [{0x327455e20} + 0xe4]  (0x35d051f28 0x35d052094) [0x12ff96a80 - Unity Child Domain]
#25 0x0000035d039648 in  UnityEditor.Handles:DrawCameraImpl (UnityEngine.Rect,UnityEngine.Camera,UnityEditor.DrawCameraMode,bool,UnityEditor.DrawGridParameters,bool,bool,bool,UnityEngine.GameObject[]) [{0x327455c88} + 0x6a8] [/Users/bokken/build/output/unity/unity/Editor/Mono/Handles.cs :: 1594u] (0x35d038fa0 0x35d0397bc) [0x12ff96a80 - Unity Child Domain]
#26 0x0000035d038f28 in  UnityEditor.Handles:DrawCameraStep1 (UnityEngine.Rect,UnityEngine.Camera,UnityEditor.DrawCameraMode,UnityEditor.DrawGridParameters,bool,bool) [{0x327455580} + 0x188] [/Users/bokken/build/output/unity/unity/Editor/Mono/Handles.cs :: 1617u] (0x35d038da0 0x35d038f4c) [0x12ff96a80 - Unity Child Domain]
#27 0x0000035d033ee0 in  UnityEditor.SceneView:DoDrawCamera (UnityEngine.Rect,UnityEngine.Rect,bool&) [{0x1276a3100} + 0xdc8] [/Users/bokken/build/output/unity/unity/Editor/Mono/SceneView/SceneView.cs :: 2096u] (0x35d033118 0x35d0347dc) [0x12ff96a80 - Unity Child Domain]
#28 0x0000035d00cad4 in  UnityEditor.SceneView:DoOnGUI () [{0x1276a32b8} + 0x1254] [/Users/bokken/build/output/unity/unity/Editor/Mono/SceneView/SceneView.cs :: 2415u] (0x35d00b880 0x35d00e1e4) [0x12ff96a80 - Unity Child Domain]
#29 0x0000035d00b814 in  UnityEditor.SceneView:OnSceneGUI () [{0x1276a3290} + 0x54] [/Users/bokken/build/output/unity/unity/Editor/Mono/SceneView/SceneView.cs :: 2328u] (0x35d00b7c0 0x35d00b830) [0x12ff96a80 - Unity Child Domain]
#30 0x0000035c4cb5e8 in  UnityEngine.UIElements.IMGUIContainer:DoOnGUI (UnityEngine.Event,UnityEngine.Matrix4x4,UnityEngine.Rect,bool,UnityEngine.Rect,System.Action,bool) [{0x17bfb3008} + 0xe50] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs :: 355u] (0x35c4ca798 0x35c4ccbd8) [0x12ff96a80 - Unity Child Domain]
#31 0x0000035c4a8dc8 in  UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent (UnityEngine.Event,UnityEngine.Matrix4x4,UnityEngine.Rect,System.Action,bool) [{0x17bfb3210} + 0x6d8] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs :: 676u] (0x35c4a86f0 0x35c4a9104) [0x12ff96a80 - Unity Child Domain]
#32 0x0000035ceeaf24 in  UnityEngine.UIElements.IMGUIContainer:DoIMGUIRepaint () [{0x17bfb3080} + 0x514] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs :: 536u] (0x35ceeaa10 0x35ceeafa8) [0x12ff96a80 - Unity Child Domain]
#33 0x0000035cee86bc in  UnityEngine.UIElements.UIR.RenderChainCommand:ExecuteNonDrawMesh (UnityEngine.UIElements.UIR.DrawParams,single,System.Exception&) [{0x2ec07fff8} + 0x6bc] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/Renderer/UIRenderer/UIRenderers.cs :: 130u] (0x35cee8000 0x35cee9c1c) [0x12ff96a80 - Unity Child Domain]
#34 0x0000035ced6c00 in  UnityEngine.UIElements.UIR.UIRenderDevice:EvaluateChain (UnityEngine.UIElements.UIR.RenderChainCommand,UnityEngine.Material,UnityEngine.Material,UnityEngine.Texture,UnityEngine.Texture,single,Unity.Collections.NativeSlice`1<UnityEngine.UIElements.UIR.Transform3x4>,Unity.Collections.NativeSlice`1<UnityEngine.Vector4>,UnityEngine.MaterialPropertyBlock,bool,System.Exception&) [{0x2ec07f978} + 0x1710] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/Renderer/UIRenderer/UIRenderDevice.cs :: 1003u] (0x35ced54f0 0x35ced71e4) [0x12ff96a80 - Unity Child Domain]
#35 0x0000035ced1c40 in  UnityEngine.UIElements.UIR.RenderChain:Render () [{0x2ec0ee2a0} + 0x738] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/Renderer/UIRRenderChain.cs :: 500u] (0x35ced1508 0x35ced1ddc) [0x12ff96a80 - Unity Child Domain]
#36 0x0000035cdad0fc in  UnityEngine.UIElements.UIRRepaintUpdater:Update () [{0x328058bb0} + 0x474] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/Renderer/UIRRepaintUpdater.cs :: 79u] (0x35cdacc88 0x35cdad13c) [0x12ff96a80 - Unity Child Domain]
#37 0x0000035c45c04c in  UnityEngine.UIElements.VisualTreeUpdater:UpdateVisualTreePhase (UnityEngine.UIElements.VisualTreeUpdatePhase) [{0x2ec1b9880} + 0x174] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/VisualTreeUpdater.cs :: 111u] (0x35c45bed8 0x35c45c0c0) [0x12ff96a80 - Unity Child Domain]
#38 0x0000035cd30368 in  UnityEngine.UIElements.Panel:UpdateForRepaint () [{0x2b2fd8668} + 0x110] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/Panel.cs :: 1012u] (0x35cd30258 0x35cd30390) [0x12ff96a80 - Unity Child Domain]
#39 0x0000035cd2d8e0 in  UnityEngine.UIElements.Panel:Repaint (UnityEngine.Event) [{0x2ccc005f0} + 0x330] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/Panel.cs :: 1073u] (0x35cd2d5b0 0x35cd2d9e8) [0x12ff96a80 - Unity Child Domain]
#40 0x0000035c45b100 in  UnityEngine.UIElements.UIElementsUtility:DoDispatch (UnityEngine.UIElements.BaseVisualElementPanel) [{0x2ec1b97a0} + 0x278] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs :: 454u] (0x35c45ae88 0x35c45b760) [0x12ff96a80 - Unity Child Domain]
#41 0x0000035c45ab5c in  UnityEngine.UIElements.UIElementsUtility:UnityEngine.UIElements.IUIElementsUtility.ProcessEvent (int,intptr,bool&) [{0x2b2f044a0} + 0x1b4] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs :: 205u] (0x35c45a9a8 0x35c45abf8) [0x12ff96a80 - Unity Child Domain]
#42 0x0000035c45a6ac in  UnityEngine.UIElements.UIEventRegistration:ProcessEvent (int,intptr) [{0x2ec1b96e0} + 0x194] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs :: 74u] (0x35c45a518 0x35c45a824) [0x12ff96a80 - Unity Child Domain]
#43 0x0000035c45a488 in  UnityEngine.UIElements.UIEventRegistration/<>c:<.cctor>b__1_2 (int,intptr) [{0x2b2f050c8} + 0x70] [/Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs :: 28u] (0x35c45a418 0x35c45a4c4) [0x12ff96a80 - Unity Child Domain]
#44 0x0000035c45a1a0 in  UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) [{0x1276027a0} + 0xe0] [/Users/bokken/build/output/unity/unity/Modules/IMGUI/GUIUtility.cs :: 189u] (0x35c45a0c0 0x35c45a1f4) [0x12ff96a80 - Unity Child Domain]
#45 0x0000035c45a2e8 in  (wrapper runtime-invoke) <Module>:runtime_invoke_void_int_intptr_intptr& (object,intptr,intptr,intptr) [{0x2ec449450} + 0xf0]  (0x35c45a1f8 0x35c45a3d4) [0x12ff96a80 - Unity Child Domain]
#46 0x0000016b094884 in mono_jit_runtime_invoke
#47 0x0000016b21b480 in do_runtime_invoke
#48 0x0000016b21b3a0 in mono_runtime_invoke
#49 0x0000010103fc34 in scripting_method_invoke(ScriptingMethodPtr, ScriptingObjectPtr, ScriptingArguments&, ScriptingExceptionPtr*, bool)
#50 0x0000010101ee40 in ScriptingInvocation::Invoke(ScriptingExceptionPtr*, bool)
#51 0x0000010115429c in Scripting::UnityEngine::GUIUtilityProxy::ProcessEvent(int, void*, bool*, ScriptingExceptionPtr*)
#52 0x00000101a4162c in GUIView::ProcessRetainedMode(InputEvent&, GUIState&, int)
#53 0x00000102e39850 in GUIView::OnInputEvent(InputEvent&)
#54 0x00000101a40ae4 in GUIView::ProcessInputEvent(InputEvent&)
#55 0x00000102e48ba8 in -[GUIRenderView renderRect:]
#56 0x00000102e46694 in -[GUIMetalView drawRect:]
#57 0x00000102e3b29c in GUIView::DoPaint()
#58 0x00000102e3b548 in GUIView::RepaintAll(bool)
#59 0x00000101b9f134 in PlayerLoopController::InternalUpdateScene(bool, bool)
#60 0x00000101b9779c in PlayerLoopController::UpdateSceneIfNeededFromMainLoop()
#61 0x00000101b94e18 in Application::TickTimer()
#62 0x00000102e26c94 in -[EditorApplication TickTimer]
#63 0x000001b08fb44c in __NSFireTimer
#64 0xba498001af9d354c in __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__
#65 0x4b6e8001af9d31f4 in __CFRunLoopDoTimer
#66 0x3000001af9d2d4c in __CFRunLoopDoTimers
#67 0x8d6c8001af9b8734 in __CFRunLoopRun
#68 0xec770001af9b7878 in CFRunLoopRunSpecific
#69 0x2b3c8001b9097fa0 in RunCurrentEventLoopInMode
#70 0xc7020001b9097de4 in ReceiveNextEventCommon
#71 0x21400001b9097b2c in _BlockUntilNextEventMatchingListInModeWithFilter
#72 0x4b210001b2c3d84c in _DPSNextEvent
#73 0xf9620001b2c3c9dc in -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]
#74 0x8671800102e26aec in -[EditorApplication checkForFocusChange:]
#75 0x000001b08fb44c in __NSFireTimer
#76 0xa5640001af9d354c in __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__
#77 0x9d490001af9d31f4 in __CFRunLoopDoTimer
#78 0xb0048001af9d2d4c in __CFRunLoopDoTimers
#79 0xa8748001af9b8734 in __CFRunLoopRun
#80 0x7e570001af9b7878 in CFRunLoopRunSpecific
#81 0x5f288001b9097fa0 in RunCurrentEventLoopInMode
#82 0x417d8001b9097de4 in ReceiveNextEventCommon
#83 0xa34e8001b9097b2c in _BlockUntilNextEventMatchingListInModeWithFilter
#84 0xf760001b2c3d84c in _DPSNextEvent
#85 0x7b080001b2c3c9dc in -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]
#86 0x3008001b2c30e0c in -[NSApplication run]
#87 0x141d0001b2c08250 in NSApplicationMain
#88 0xc45b800102e3f510 in EditorMain(int, char const**)
#89 0x00000102e3f84c in main
#90 0x000001af5afe50 in start

```

**Comment by crashkonijn** [comment]
Hi @codingwatching 

Thanks for checking out this project!

The log doesn't seem to contain any mention of anything from this package, I thinks this might actually be a bug in Unity.

Unfortunately I don't have access to a (M1) MacBook, Is there a change you're willing to join our [discord ](https://discord.gg/dCPnHaYNrm) so we can have a look at it together?

Thanks in advance!

**Comment by crashkonijn** [comment]
Hi @codingwatching 

I started researching a bit, and there seems to be a lot of issues with 2022.2 on MacOs atm. I specifically found [this topic](https://forum.unity.com/threads/frequent-crashes-on-apple-m1-max-w-unity-2022-2-5f1.1399063) about it.

Specifically [this post](https://forum.unity.com/threads/frequent-crashes-on-apple-m1-max-w-unity-2022-2-5f1.1399063/page-2#post-8946672) in that thread mentions crashed on ScriptableRenderContext, with a Unity dev confirming that's included in that issues. A couple post earlier the dev mentions the fix should be out for 2022.2.17 (in ~2 weeks).

Perhaps you could try and update to 2022.15 (for now) to see if that helps?
